addons/web_editor/controllers/main.py

```diff
@@ -301,130 +301,180 @@ def get_image_info(self, image_id=None, xml_id=None):
-            scss = get_scss and scss_files_data_by_bundle or [],
+            views=get_views and views or [],
+            scss=get_scss and scss_files_data_by_bundle or [],
+            js=get_js and js_files_data_by_bundle or [],
         )
 
-    def save_scss_view_hook(self):
+    def load_resources(self, file_type, views, bundles_restriction, only_user_custom_files):
+        files_data = []
+        resources_type_info = {'t_call_assets_attribute': 't-js', 'mimetype': 'text/javascript'}
+        if file_type == 'scss':
+            resources_type_info = {'t_call_assets_attribute': 't-css', 'mimetype': 'text/scss'}
+
+        # Load already customized files attachments
+        custom_url = self._make_custom_scss_or_js_file_url("%%.%%", "%%")
+        custom_attachments = self.get_custom_attachment(custom_url, op='=like')
+
+        # Compile regex outside of the loop
+        # This will used to exclude library scss files from the result
+        excluded_url_matcher = re.compile("^(.+/lib/.+)|(.+import_bootstrap.+\.scss)$")
+
+        # First check the t-call-assets used in the related views
+        url_infos = dict()
+        for v in views:
+            for asset_call_node in etree.fromstring(v["arch"]).xpath("//t[@t-call-assets]"):
+                if asset_call_node.get(resources_type_info['t_call_assets_attribute']) == "false":
+                    continue
+                asset_name = asset_call_node.get("t-call-assets")
+
+                # Loop through bundle files to search for file info
+                _files_data = []
+                for file_info in request.env["ir.qweb"]._get_asset_content(asset_name, {})[0]:
+                    if file_info["atype"] != resources_type_info['mimetype']:
+                        continue
+                    url = file_info["url"]
+
+                    # Exclude library files (see regex above)
+                    if excluded_url_matcher.match(url):
+                        continue
+
+                    # Set files URL filters
+                    custom_files = [
+                        '/website/static/src/scss/user_custom_bootstrap_overridden.scss',
+                        '/website/static/src/scss/user_custom_rules.scss',
+                        '/website/static/src/js/user_custom_javascript.js'
+                    ]
+                    url_from_custom_files = url in custom_files
+
+                    # Check if the file is customized and get bundle/path info
+                    scss_or_js_file_data = self._match_scss_or_js_file_url(url)
+                    if not scss_or_js_file_data:
+                        continue
+                    file_is_customized = scss_or_js_file_data['customized'] is True
+
+                    # Save info according to the filter (arch will be fetched later)
+                    url_infos[url] = scss_or_js_file_data
+                    if file_type == 'scss':
+                        if only_user_custom_files and len(bundles_restriction) > 0:
```

Comment:
Using bundle restriction should not be done here. We filter after the loop.

Reply:
_Last TODO_